src/github.com/getlantern/idletiming/idletiming_conn.go

```diff
@@ -100,12 +105,13 @@ func (c *IdleTimingConn) Read(b []byte) (int, error) {
 			n, err := c.conn.Read(b)
 			c.markActive(n)
 			totalN = totalN + n
-			timedOut := isTimeout(err)
-			if timedOut {
-				// Ignore timeouts when using deadline based on IdleTimeout
+			hitMaxDeadline := isTimeout(err) && !time.Now().Before(maxDeadline)
```

Comment:
Yeah, ultimately we don't have control over the underlying connection and what's done with it.  Really, all these changes are just me trying to nitpick the heck out of this code to see if I can get it to a point where it doesn't break in production.  I never was able to reproduce in unit testing.

Note - fl-singapore-004-1 has been running spike free for about 23 minutes now, which is not bad relative to previous tests.  If it still looks good in the morning, I'll push the updated code to all fallbacks and see what happens.